lib/ansible/playbook/__init__.py

```diff
@@ -40,17 +39,18 @@
 
 class Playbook:
 
-    def __init__(self, loader):
+    def __init__(self, loader, options=None):
```

Comment:
Passing a command line options instance (a optparse.Values ) deep down into Playbook and Play objects seems wrong to me. It couples ansible.playbook.* stuff to a particular cli arg parser (things like ansible.playbook.FieldAttributeBase._set_option have to deal with the implementation details of optparse)

Reply:
possibly, this way i saw as 'least intrusive' since now we do it by 'side loading' into play_context, this keeps the correct precedence (cli sets default for keywords).

Reply:
we can change it slightly and make 'options' a dict which is populated from the cli object, but I didn't think we needed that at this point

Reply:
We've talked about converting options to a dict for 3+ years... it would be nice to finally do it.

Reply:
I don't disagree, but i would not make it a requirement here